### PR TITLE
helm: Allow setting loadBalancerSourceRanges for clustermesh-apiserver Kubernetes service

### PR DESCRIPTION
**Pull request by mantoine96**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

This PR enables users to set `loadBalancerSourceRanges` for the `clustermesh-apiserver` Kubernetes service when it is of type LoadBalancer. This is useful for users that might want to restrict access to said load balancer.

```release-note
helm: Allow setting `loadBalancerSourceRanges` for clustermesh-apiserver kubernetes service
```

### CODE REVIEW

File: install/kubernetes/cilium/templates/clustermesh-apiserver/service.yaml [hunk 1]
```diff
@@ -41,4 +41,8 @@ spec:
   {{- if or (eq .Values.clustermesh.apiserver.service.enableSessionAffinity "Always") (and (eq .Values.clustermesh.apiserver.service.enableSessionAffinity "HAOnly") (gt (int .Values.clustermesh.apiserver.replicas) 1)) }}
   sessionAffinity: ClientIP
   {{- end }}
+  {{- if and (eq "LoadBalancer" .Values.clustermesh.apiserver.service.type) .Values.clustermesh.apiserver.service.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges:
+    {{- toYaml .Values.clustermesh.apiserver.service.loadBalancerSourceRanges | nindent 4 }}
+  {{- end }}
```
Comment (giorio94) [inline install/kubernetes/cilium/templates/clustermesh-apiserver/service.yaml]: Nit. I'd suggest moving this setting close to the other LoadBalancer related ones, for consistency.

File: install/kubernetes/cilium/values.yaml.tmpl [hunk 2]
```diff
@@ -3028,6 +3028,13 @@ clustermesh:
       # Allows to configure a specific loadBalancerIP on the clustermesh-apiserver
       # LB service in case the Service type is set to LoadBalancer.
       loadBalancerIP: ~
+      # @schema
+      # type: [array]
+      # @schema
```
Comment (giorio94) [inline install/kubernetes/cilium/values.yaml.tmpl]: Nit. Looks like this is inferred automatically, and could be omitted.

**Review by giorio94 — APPROVED** [review]
Thanks! A couple of non-blocking nits inline.

File: install/kubernetes/cilium/templates/clustermesh-apiserver/service.yaml (hunk 1 above)
Comment (mantoine96) [reply install/kubernetes/cilium/templates/clustermesh-apiserver/service.yaml]: Thank you for this. I've moved it up to be just after `LoadBalancerIp` so the LoadBalancer attributes are now close to one another + sorted alphabetically.

File: install/kubernetes/cilium/values.yaml.tmpl (hunk 2 above)
Comment (mantoine96) [reply install/kubernetes/cilium/values.yaml.tmpl]: Cool, removed :)

**Comment by mantoine96** [comment]
Sorry @giorio94 I had to push again to address a failed Helm smoke test. Would you mind starting the tests again?

**Review by qmonnet — APPROVED** [review]
Thanks!

**Review by aanm — APPROVED** [review]

**Pull request merged**